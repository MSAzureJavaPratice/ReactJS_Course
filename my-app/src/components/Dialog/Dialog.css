.dialog-overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.5);
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 1000;
}

.dialog-content {
    background: black;
    border-radius: 0px;
    width: 700px;
    padding: 20px;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.3);
    border: solid whitesmoke;
}

.dialog-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background: black;
    border-radius: 0px;
    padding: 15px;
    border: solid whitesmoke;
}

.dialog-header h2 {
    color: red;
    /* Ensure the title text is white */
    margin: 0;
    /* Optional: remove margin for a cleaner look */
}

.dialog-close {
    background: transparent;
    border: none;
    font-size: 1.5rem;
    cursor: pointer;
}

.dialog-body {
    margin-top: 20px;
}