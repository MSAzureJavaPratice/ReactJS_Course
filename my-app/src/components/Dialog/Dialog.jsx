import React from "react";
import { Portal } from 'react-portal';
import FocusTrap from 'focus-trap-react';
import './Dialog.css';

const Dialog = ({ isOpen, onClose, title, children }) => {
  if (!isOpen) return null;

  return (
    <Portal>
      <div className="dialog-overlay" onClick={onClose}>
        <FocusTrap>
          <div className="dialog" onClick={(e) => e.stopPropagation()}>
            <header className="dialog-header">
              <h2>{title}</h2>
              <button onClick={onClose} aria-label="Close">
                Ã—
              </button>
            </header>
            <div className="dialog-content">{children}</div>
          </div>
        </FocusTrap>
      </div>
    </Portal>
  );
};

export default Dialog;
