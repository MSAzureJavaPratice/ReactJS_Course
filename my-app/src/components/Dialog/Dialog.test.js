import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Dialog from "./Dialog";

describe("Dialog Component", () => {
  const onClose = jest.fn();

  test("renders dialog with title and children", () => {
    render(
      <Dialog title="Test Dialog" onClose={onClose}>
        <p> Dialog body content </p>{" "}
      </Dialog>
    );

    expect(screen.getByText("Test Dialog")).toBeInTheDocument();
    expect(screen.getByText("Dialog body content")).toBeInTheDocument();
  });

  test("calls onClose when close button is clicked", () => {
    render(<Dialog title="Test Dialog" onClose={onClose} />);
    fireEvent.click(screen.getByText("Ã—"));
    expect(onClose).toHaveBeenCalled();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <Dialog title="Snapshot Test" onClose={onClose}>
        <p> Dialog body content </p>{" "}
      </Dialog>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
