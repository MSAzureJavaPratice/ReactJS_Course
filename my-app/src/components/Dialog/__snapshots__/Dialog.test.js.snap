// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component matches snapshot 1`] = `<DocumentFragment />`;

exports[`Dialog Component matches snapshot when closed 1`] = `<DocumentFragment />`;
